server/controllers/finance/transactions.js

```diff
@@ -48,6 +47,30 @@ function parseDocumentMapString(text) {
   return safeDeletionMethods[key];
 }
 
+/**
+ * @functio isUserAuthorized
+ *
+ * @description
+ * This function checks if the user is authorized to delete the record from their "actions"
+ * set by their user role.
+ *
+ * @returns {Boolean} true if the user is authorized, false if not.
+ */
+function isUserAuthorized(text, actions) {
+  const DELETE_CASH_PAYMENT = 3;
+  const DELETE_INVOICE = 4;
+  const DELETE_VOUCHER = 7;
```

Comment:
Again, shouldn't these be from bhConstants?

Reply:
bhConstants is a client-side script, so no.  We probably should assemble these from a constant script on the server.

https://github.com/IMA-WorldHealth/bhima/issues/1221 requests that we build out all our constants from a central location.  This will require a pretty significant refactor.  I'll make this change the start, but we can later go through and update account type ids, etc, that are hard-coded.